pkg/adapter/main.go

```diff
@@ -49,9 +48,11 @@ type Adapter interface {
 type AdapterConstructor func(ctx context.Context, env EnvConfigAccessor, client cloudevents.Client, reporter source.StatsReporter) Adapter
 
 func Main(component string, ector EnvConfigConstructor, ctor AdapterConstructor) {
-	flag.Parse()
+	MainWithContext(signals.NewContext(), component, ector, ctor)
+}
 
-	ctx := signals.NewContext()
+func MainWithContext(ctx context.Context, component string, ector EnvConfigConstructor, ctor AdapterConstructor) {
+	flag.Parse()
```

Comment:
I think `Parse()` cannot be called multiple times, so it will break code calling directly MainWithContext. Just make sure it's not called or check if flags have already been parsed

Reply:
In a source you use or one main (the one that we have called `adapter.Main`) or the new one with messages (called `adapter.MainMessageAdapter`), not both at the same time

Reply:
just for adapter.MainWithContext

Reply:
WDYM?